arthur_bench/scoring/summary_quality.py

```diff
@@ -59,8 +59,8 @@ class SummaryQuality(Scorer):
     Comprehensive measure of summarization quality compared to a reference summary.
     """
 
-    def __init__(self):
-        self.evaluator = LLMChain(llm=ChatOpenAI(temperature=0), prompt=COMPARE)
+    def __init__(self, llm = ChatOpenAI(temperature=0)):
```

Comment:
hi @maccam912, thanks so much for opening this PR! We're definitely focused on improving the configurability of evaluators in the future. A couple of things to call out with this change:

1) Differing context windows for other LLMs. This scorer performs [input truncation](https://github.com/arthur-ai/bench/blob/main/arthur_bench/scoring/summary_quality.py#L28) according to gpt-3.5-turbo's context window, so we'll want to also update that logic

2) The summary prompt in its current state requires a chat model. We'll need to add formatting to the prompt or validation that the llm passed is a chat model in the meantime

Reply:
For 1, I added arguments for context window size and encoder, with defaults being fine for this chat model. I added a line to log a warning if the user overrides the LLM with one that isn't the ChatOpenAI class. If I am passing my own SummaryScorer I don't mind needing to double check those things, and default behavior throughout should be the same as before. How do those changes look?